explorer/nodewatch/puller/model/NodeInfo.py

```diff
@@ -0,0 +1,68 @@
+class NodeInfo:
+	"""Database node model."""
+
+	# Role flag indicating an API node.
+	API_ROLE_FLAG = 2
+
+	def __init__(self, network_id, network_generation_hash_seed, main_public_key, node_public_key, endpoint, name, version, roles):
+		"""Creates a descriptor."""
+
+		# pylint: disable=too-many-arguments
+
+		self.network_id = network_id
+		self.network_generation_hash_seed = network_generation_hash_seed
+		self.main_public_key = main_public_key
+		self.node_public_key = node_public_key
+		self.endpoint = endpoint
+		self.name = name
+		self.version = version
+		self.roles = roles
+
+	@property
+	def has_api(self):
+		"""Returns true if this node supports a REST API."""
+
+		return bool(self.roles & self.API_ROLE_FLAG)
+
+	def to_dict(self):
+		"""Formats the node descriptor as a dictionary."""
+
+		# TODO: should we include network identification in here?
+
+		return {
+			'mainPublicKey': str(self.main_public_key),
+			'nodePublicKey': str(self.node_public_key) if self.node_public_key else None,
+			'endpoint': self.endpoint,
+			'name': self.name,
+			'version': self.version,
+			'roles': self.roles
+		}
+
+	def __eq__(self, other):
+		# pylint: disable=protected-access
+		return isinstance(other, NodeInfo) and (
+			self.network_id == other.network_id and (
+				self.network_generation_hash_seed == other.network_generation_hash_seed and (
+					self.main_public_key == other.main_public_key and (
+						self.node_public_key == other.node_public_key and (
+							self.endpoint == other.endpoint and (
+								self.name == other.name and (
```

Comment:
do we need/want name to actually be equal? (esp if everything else matches)

Reply:
i don't mind, we can probably remove that